src/plugins/discover/public/application/main/components/layout/use_discover_histogram.ts

```diff
@@ -214,6 +235,26 @@ export const useDiscoverHistogram = ({
     unifiedHistogram?.setBreakdownField(breakdownField);
   }, [breakdownField, unifiedHistogram]);
 
+  const columns = useAppStateSelector((state) => state.columns);
+  const documentState = useDataState(savedSearchData$.documents$);
+  useEffect(() => {
+    // Update the columns only when the query changes
+    if (!isEqual(columns, prev.current.columns) && !isEqual(query, prev.current.query)) {
+      unifiedHistogram?.setColumns(columns);
+      prev.current.query = query;
+      prev.current.columns = columns;
+    }
+  }, [columns, query, unifiedHistogram]);
+
+  useEffect(() => {
```

Comment:
I am afraid that I don't understand this suggestion. I would appreciate some help here :)

Reply:
This will be fixed in my PR -- I added support for getting total hits from the Lens response as well as sending a separate request for total hits in text based mode when the chart is hidden.